lectures/supply_demand_multiple_goods.md

```diff
@@ -422,13 +602,273 @@ $$
+
+        return c, p
+
+    def equilibrium_with_monopoly(self):
+        """
+        Compute the equilibrium price and allocation when there is a monopolist supplier
+        """
+        Pi, b, h, mu, J = self.Pi, self.b, self.h, self.mu, self.J
+        H = .5 * (J + J.T)
+
+        # allocation
+        q = inv(mu * H + 2 * Pi.T @ Pi) @ (Pi.T @ b - mu * h)
+
+        # price
+        p = 1 / mu * (Pi.T @ b - Pi.T @ Pi @ q)
+
+        if any(Pi @ q - b >= 0):
+            raise Exception('invalid result: set bliss points further away')
+
+        return q, p
+
+    def compute_surplus(self):
+        """
+        Compute consumer and producer surplus for single good case
+        """
+        if self.n != 1:
+            raise Exception('not single good')
+        h, J, Pi, b, mu = self.h.item(), self.J.item(), self.Pi.item(), self.b.item(), self.mu
+        H = J
+
+        # supply/demand curve coefficients
+        s0, s1 = h, H
+        d0, d1 = 1 / mu * Pi * b, 1 / mu * Pi**2
+
+        # competitive equilibrium
+        c, p = self.competitive_equilibrium()
+
+        # calculate surplus
+        c_surplus = d0 * c - .5 * d1 * c**2 - p * c
+        p_surplus = p * c - s0 * c - .5 * s1 * c**2
+
+        return c_surplus, p_surplus
+```
+
+Then define a function that plots demand and supply curves and labels surpluses and equilibrium.
+
+```{code-cell} ipython3
+def plot_competitive_equilibrium(PE):
+    """
+    Plot demand and supply curves, producer/consumer surpluses, and equilibrium for
+    a single good production economy
+
+    Args:
+        PE (class): A initialized production economy class
+    """
+    # get singleton value
+    J, h, Pi, b, mu = PE.J.item(), PE.h.item(), PE.Pi.item(), PE.b.item(), PE.mu
+    H = J
+
+    # compute competitive equilibrium
+    c, p = PE.competitive_equilibrium()
+    c, p = c.item(), p.item()
+
+    # inverse supply/demand curve
+    supply_inv = lambda x: h + H * x
+    demand_inv = lambda x: 1 / mu * (Pi * b - Pi * Pi * x)
+
+    xs = np.linspace(0, 2 * c, 100)
+    ps = np.ones(100) * p
+    supply_curve = supply_inv(xs)
+    demand_curve = demand_inv(xs)
+
+    # plot
+    plt.figure()
+    plt.plot(xs, supply_curve, label='Supply', color='#020060')
+    plt.plot(xs, demand_curve, label='Demand', color='#600001')
+
+    plt.fill_between(xs[xs <= c], demand_curve[xs <= c], ps[xs <= c], label='Consumer surplus', color='#EED1CF')
+    plt.fill_between(xs[xs <= c], supply_curve[xs <= c], ps[xs <= c], label='Producer surplus', color='#E6E6F5')
+
+    plt.vlines(c, 0, p, linestyle="dashed", color='black', alpha=0.7)
+    plt.hlines(p, 0, c, linestyle="dashed", color='black', alpha=0.7)
+    plt.scatter(c, p, zorder=10, label='Competitive equilibrium', color='#600001')
+
+    plt.legend(loc='upper right')
+    plt.margins(x=0, y=0)
+    plt.ylim(0)
+    plt.xlabel('Quantity')
+    plt.ylabel('Price')
+    plt.show()
+```
+
+#### Example: Single Agent with One Good and with Production
```

Comment:
These examples, which are for single agent and single good, could be linked back to ``supply_demand_intro``.

Reply:
Good idea, only if you have time.

Reply:
> Change capitalization of section titles: https://manual.quantecon.org/styleguide/writing.html#titles-and-headings

Thanks @jstac . Done.